### ISSUE TITLE
YTS IMDB LINK

### ISSUE THREAD

**Issue by melbasuonyy** [makkoncept/yts.am_scraper_#1]
Hi It's a great tool and helped a lot but if you can add IMDB URL so it will be easy to compare the data if you have huge library 

Thanks

**Comment by makkoncept** [comment]
hey @melbasuonyy , thank you for appreciating the project. Can you elaborate more on the feature. You want the imdb link to be one of the data scraped and added to the csv?

**Comment by melbasuonyy** [comment]
exactly, it will be  easy to track the movies as IMDB has it's unique number for each movie tt*****

**Comment by makkoncept** [comment]
@melbasuonyy  i have added the feature, you can run the script and have a look.

**Comment by melbasuonyy** [comment]
I tried the link but it not fetching links for movies which include letters like ( & , : ) if you can fix it it will be great

**Comment by makkoncept** [comment]
@melbasuonyy 
It is working on my machine !
Some columns are left blank as there is no link on the site in the first place.

![test_run](https://user-images.githubusercontent.com/34679965/41197870-704c8aa2-6c88-11e8-98e2-a89924804084.png)

### TroubleShooting

- Have you cloned the updated version of the script? 
- Maybe the site is blocked at your place, I use a proxy to get around this.
- You can test run the script at few pages initially, In the above example i have ran it on on the first page only. You can do this by changing the range in `for page in range(1, 357):` in the code.

If the problem is still not solved, let me know.

**Comment by melbasuonyy** [comment]
yes it's same on my machine as well the links are grabbed  for the same movies as well i think movies with special characters are showing as blank (check raw 3 / 15 / 21) but it's manageable thanks for your work

I have one more suggestion you can look into it if it will help you can also include the download links for the torrents on the file with each movie it will be easy for people who wants to download batch files one time

**Comment by makkoncept** [comment]
Yeah , there are many edge cases and i think searching and writing the condition for all of them would make the code messy. Also after a month I came to know that yts has an API :p .

I previously taught about it but as you can see the links are not uniform on the site. I think adding only 720p and 1080p rows(as 3D is not available for majority of films) would be good and if any torrent is not available, leave the cell blank.

**Comment by makkoncept** [comment]
@melbasuonyy i have added the feature, you can run the script and have a look.

**Comment by makkoncept** [comment]
@melbasuonyy  , I have made some enhancement to the script and it can now handle special characters in the movie name. I have not come across an exception but it can exist if the movie has a very bizarre name. Let me know if you found one.

![final](https://user-images.githubusercontent.com/34679965/41202766-4746d320-6ceb-11e8-9c76-d729531eed19.png)

comparing this with the above output, movie names with special characters are also scraped here.

**Comment by melbasuonyy** [comment]
![untitled](https://user-images.githubusercontent.com/29347988/41205831-a7a4dfb2-6d0b-11e8-8b03-05a6eb19e19c.png)

i'm getting the below error

**Comment by melbasuonyy** [comment]
I have done the full site and only 95 movies are not scraped which is great so overall it scraped all the links expect 2 % 

Thanks alot

**Comment by makkoncept** [comment]
@melbasuonyy 
You can scrape the whole website by updating the range in `for page in range(1, 357):` to `for page in range(1,383):`  in the script as now new torrents are added and the total number of pages are 382 .


![page](https://user-images.githubusercontent.com/34679965/41223671-1219129c-6d88-11e8-84dc-0f9ebccefd4c.png)

Welcome :)
